### PR TITLE
Add optimistic updates section in the useMutation docs

### PR DESCRIPTION
**Pull request by shreyas44**
Also, you can distinguish that the data is real or optimistic through the isLoading property. 
When **isLoading is true**, **data is optimistic** otherwhise when **isloading is false** the **data is real**.

_Originally posted by @morrys in https://github.com/relay-tools/relay-hooks/issues/145#issuecomment-770204208_

- Make the above stated point clear in the `useMutation` docs
- Add a section about Optimistic UI

### CODE REVIEW

File: docs/useMutation.md [hunk 1]
```diff
@@ -100,6 +100,15 @@ The `mutationState` object has the following properties:
 - `data`: the response data for the mutation
 - `error`: any errors returned by the mutation
 
+## Optimistic Updates
+
+To make your UI more responsive you can use the Optimistic UI pattern. You can read more about Optimistic UI in the [Relay Docs](https://relay.dev/docs/en/mutations#optimistic-updates) or the [Apollo Docs](https://www.apollographql.com/docs/react/performance/optimistic-ui/).
+
+An Optimistic Response can be provided via the `optimisticResponse` parmater of the mutation options.
```
Comment (morrys) [inline docs/useMutation.md:107]: here change parmater to parameter 👍

File: docs/useMutation.md [hunk 2]
```diff
@@ -100,6 +100,15 @@ The `mutationState` object has the following properties:
 - `data`: the response data for the mutation
 - `error`: any errors returned by the mutation
 
+## Optimistic Updates
```
Comment (morrys) [inline docs/useMutation.md:103]: add this section also in this file:
https://github.com/relay-tools/relay-hooks/blob/master/useMutation.md

**Review by morrys — CHANGES_REQUESTED** [review]
hi @shreyas44,
in the meantime, thanks for your contribution :)

I've added a couple of comments to your PR

File: docs/useMutation.md (hunk 2 above)
Comment (shreyas44) [reply docs/useMutation.md:103]: Quick question, why maintain 2 copies of the same files? 

The links in the README could link to the files in the `docs` folder to make it less redundant probably?

Comment (morrys) [reply docs/useMutation.md:103]: @shreyas44 great suggestion;)
can you do it in this PR? ;)

Comment (shreyas44) [reply docs/useMutation.md:103]: I think it would be better to open a separate PR as it's for a different purpose?

Comment (morrys) [reply docs/useMutation.md:103]: you're right but I didn't want to bother you doing extra work

Comment (shreyas44) [reply docs/useMutation.md:103]: Ahh, it's no problem! Just created a new PR for this #157 🙂

**Pull request merged**